### ISSUE TITLE
Unlocking should play a scale effect animation

### ISSUE THREAD

**Issue by hanaral** [elementary/gala#1040]
## Problem

Unlocking right now jarringly just flashes a black screen the moment you unlock, which is disorientating and doesn't utilise the idea that there is a gray 'system layer' beneath everything.

## Proposal

When the user unlocks their device Gala could do a nice fade out of the lockscreen cards and wingpanel, and then a smooth zoom-scale effect of the desktop filling the screen (somewhat like entering a workspace from the overview)

## Prior Art

I rendered out what the transition could look like:

https://user-images.githubusercontent.com/58219504/106654310-e1ebfc80-658f-11eb-9de4-cd2f3673d12e.mov

**Comment by cassidyjames** [comment]
Honestly if we could swing it I would love for the wallpaper on the user's card to scale up to fill the screen, then the dock and panel to slide in from the edges; that was one of the original spatial designs, and is part of why the greeter has the same "system layer" beneath it as the multitasking view.

**Comment by hanaral** [comment]
> Honestly if we could swing it I would love for the wallpaper on the user's card to scale up to fill the screen, then the dock and panel to slide in from the edges

I was going to actually open the issue based on this originally but I thought it might be a little much in effort and in overkill-ness. However it's still probably something to aim for 🚀️

**Comment by cassidyjames** [comment]
Quick and dirty mockup


https://user-images.githubusercontent.com/611168/106661268-3d04fb80-655e-11eb-8a59-882e924a9a66.mp4

**Comment by hanaral** [comment]
Since we are going down the route of overkill, how about a quick animation of the user picture badge disappearing and the card vertically 'folding' to hide the bottom half so that it becomes a card of the user background >
the background card then quickly scales up >
it becomes the same desktop card as in the workspace overview >
it continues to grow and then the dock and wingpanel slide in 
This is probably a lot of work, but again it's something to aim for

**Comment by mcclurgm** [comment]
That looks kinda cool. Some thoughts though:

How would that deal with unlocking a full desktop? In the extreme case I know lots of people work with maximized windows, so there would have to be some major transition in there too. But for people like me who use a bunch of non-maximized windows it would be an issue too, unless you're the type who only uses one little window in the center. Unless you're thinking of this only for first login, then I suppose this is a moot point.

Also, that's an awful big zoom if the wallpaper image is going to stay the size it is right now. tbh it almost doesn't look like the desktop was zooming from the photo because it reads a bit too much like it was just going from zero size at the center of the screen. It also seems like a fair bit of visual noise for every login (unless it's only fresh logins, again). Especially if it were to zoom from an index card to a big monitor that's a big change.

**Comment by cassidyjames** [comment]
@mcclurgm 

>How would that deal with unlocking a full desktop? … Unless you're thinking of this only for first login, then I suppose this is a moot point.

Yeah I was just thinking for initial login. Not sure about locking/unlocking.

>Also, that's an awful big zoom if the wallpaper image is going to stay the size it is right now…

See https://github.com/elementary/greeter/issues/303

**Comment by hanaral** [comment]
I’d still think that having a simpler animation like in my original mock-up would go well with the unlocking side. Unless there is a reason for not using it that I haven’t realised, wouldn’t it be O.K on virtually all display sizes?

**Comment by cassidyjames** [comment]
@hanaral that would work, and would be an improvement. I just wanted to share some context on how I had pictured the design working when we designed the greeter.
